src/lib/hooks/gun/useGunProfile.js

```diff
@@ -0,0 +1,55 @@
+import { useEffect, useRef, useState } from 'react'
+import { get, result } from 'lodash'
+import userStorage from '../../gundb/UserStorage'
+import isEmail from '../../validators/isEmail'
+import isMobilePhone from '../../validators/isMobilePhone'
+import logger from '../../logger/pino-logger'
+
+const log = logger.child({ from: 'useGunProfile' })
+
+const useGunProfile = (identifier, fields = ['fullName', 'smallAvatar']) => {
+  const attr = isMobilePhone(identifier) ? 'mobile' : isEmail(identifier) ? 'email' : 'walletAddress'
+  const value = userStorage.constructor.cleanHashedFieldForIndex(attr, identifier)
```

Comment:
Static method should be called using class name from outside:
```suggestion
 // need to import UserStorageClass
  const value = UserStorageClass.cleanHashedFieldForIndex(attr, identifier)
```

inside it'ok to use `this.constructor` like 
```javascript
const self = this.constructor

self.callStaticMethod()
```
But it's not a such case